docs/reference-guide/query-modules/implement-custom-query-modules/api/python-api.md

```diff
@@ -180,457 +180,1595 @@ RETURN example.func_example("first argument");
+procedure in a query. You should not globally store the returned Vertex
+instances.
+
+**Returns**:
+
+  `Vertices` that contained in the graph.
+  
+
+**Raises**:
+
+- `InvalidContextError` - If context is invalid.
+  
+
+**Examples**:
+
+  Iteration over all graph `Vertices`.
+  
+  ```
+  graph = context.graph
+  for vertex in graph.vertices:
+  ```
+
+### is\_mutable()
+
+```python
+def is_mutable() -> bool
+```
+
+Check if the graph is mutable. Thus it can be used to modify vertices and edges.
+
+**Returns**:
+
+  A `bool` value that depends if the graph is mutable or not.
+  
+
+**Examples**:
+
+  ```graph.is_mutable()```
+
+### create\_vertex()
+
+```python
+def create_vertex() -> Vertex
+```
+
+Create an empty vertex.
+
+**Returns**:
+
+  Created `Vertex`.
+  
+
+**Raises**:
+
+- `ImmutableObjectError` - If `graph` is immutable.
+- `UnableToAllocateError` - If unable to allocate a vertex.
+  
+
+**Examples**:
+
+  Creating an empty vertex.
+  ```vertex = graph.create_vertex()```
+
+### delete\_vertex()
+
+```python
+def delete_vertex(vertex: Vertex) -> None
+```
+
+Delete a vertex if there are no edges.
+
+**Arguments**:
+
+- `vertex` - `Vertex` to be deleted
+
+**Raises**:
+
+- `ImmutableObjectError` - If `graph` is immutable.
+- `LogicErrorError` - If `vertex` has edges.
+- `SerializationError` - If `vertex` has been modified by another transaction.
+
+**Examples**:
+
+  ```graph.delete_vertex(vertex)```
+
+### detach\_delete\_vertex()
+
+```python
+def detach_delete_vertex(vertex: Vertex) -> None
+```
+
+Delete a vertex and all of its edges.
+
+**Arguments**:
+
+- `vertex` - `Vertex` to be deleted with all of its edges
+  
+
+**Raises**:
+
+- `ImmutableObjectError` - If `graph` is immutable.
+- `SerializationError` - If `vertex` has been modified by another transaction.
+
+**Examples**:
+
+  ```graph.detach_delete_vertex(vertex)```
+
+### create\_edge()
+
+```python
+def create_edge(from_vertex: Vertex, to_vertex: Vertex,
+                edge_type: EdgeType) -> None
+```
+
+Create an edge.
+
+**Arguments**:
+
+- `from_vertex` - `Vertex` from where edge is directed.
+- `to_vertex` - `Vertex&#x27;  to where edge is directed.
+- `edge_type` - `EdgeType` defines the type of edge.
+  
+
+**Raises**:
+
+- `ImmutableObjectError` - If `graph` is immutable.
+- `UnableToAllocateError` - If unable to allocate an edge.
+- `DeletedObjectError` - If `from_vertex` or `to_vertex` has been deleted.
+- `SerializationError` - If `from_vertex` or `to_vertex` has been modified by another transaction.
+
+**Examples**:
+
+  ```graph.create_edge(from_vertex, vertex, edge_type)```
+
+### delete\_edge()
+
+```python
+def delete_edge(edge: Edge) -> None
+```
+
+Delete an edge.
+
+**Arguments**:
+
+- `edge` - `Edge` to be deleted
+  
+
+**Raises**:
+
+  ImmutableObjectError if `graph` is immutable.
+  Raise SerializationError if `edge`, its source or destination vertex has been modified by another transaction.
+
```

Comment:
the errors are not in code style, and not one below the other
![image](https://user-images.githubusercontent.com/95473291/177515179-6f924a44-f8b3-4c64-b917-4c050b7dcf5b.png)